doc/architecture/decisions/0015-key-management.md

```diff
@@ -0,0 +1,222 @@
+    prompt_user_for_pin());
+
+// reboot will require passphrase re-entry
+save_to_disk(device.get_saved(prompt_user_for_passphrase()));
+```
+
+##### 4 - new box / device generated keys - connect to existing dpki identity
+
+```rust
+// generate new random seed hierarchy
+let hd = DpkiHd::new_random();
+
+// display the primary/root seed to user for backup
+display_to_user(Dpki::seed_to_mnemonic(hd.get_primary_seed()));
+
+// attach to existing identitiy
+let existing_app_pub_data = DpkiApp::fetch(
+    prompt_user_for_dpki_id());
+existing_app_pub_data.request_authorize_new_keys(hd);
+// user will need to sign the new keys into their chain from
+// another device that is already authorized
+
+// initialize a device structure with device seed number 1
+let device = DpkiDevice::new_from_seed(hd.get_device_seed(1));
+// (optional) display device seed to user for backup
+display_to_user(Dpki::seed_to_mnemonic(hd.get_device_seed(1)));
+// blink the device pin on the holoport circle
+display_to_user(device.get_device_pin());
+
+// reboot will require passphrase re-entry
+save_to_disk(device.get_saved(prompt_user_for_passphrase()));
+```
+
+##### 5 - new holo identity
+
+```rust
+// will generate a new random salt and password hash it
+// to form a seed for keypair generation
+let holo = DpkiHolo::new_random_from_passphrase(
+    prompt_user_for_passphrase());
+
+// publish on dpki dht
+let app_pub_data = DpkiApp::new_holo(
+    prompt_user_for_username(),
+    holo);
+publish_to_dpki_dht(app_pub_data);
+
+// init a device object for signing / encryption
+let device = DpkiDevice::new_from_holo(holo);
+```
+
+##### 6 - restore holo identity
+
+```rust
+// we need to fetch the distributed salt
+// in order to restore our private keys
+let username = prompt_user_for_username();
+let app_pub_data = DpkiApp::fetch(
+    lookup_dpki_id(username));
+
+// generate the same seed data / private keys from salt / passphrase
+let holo = DpkiHolo::new_from_salt_passphrase(
+    app_pub_data.get_salt_for_username(username),
+    prompt_user_for_passphrase());
+
+// init a device object for signing / encryption
+let device = DpkiDevice::new_from_holo(holo);
+```
+
+## Decision
+
+We will use the [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) standard for generating mnemonic word lists from seeds  [Rust crate is available.](https://github.com/infincia/bip39-rs)
+
+We will use [libsodium](https://download.libsodium.org/doc/) as our primary library for encryption, decryption, signatures, password hashing, etc..
+
+We will use this api sketch as a starting point for implementation in the `holochain_api` crate, i.e. for use by container/composer developers:
+
+```rust
+/// utilities for working with mnemonic word lists
+pub mod Dpki {
+    fn seed_to_mnemonic(seed: &[u8; 32]) -> String;
+    fn mnemonic_to_seed(mnemonic: &str) -> [u8; 32];
+}
+
+/// struct for hd seed derivation handling
+impl DpkiHd {
+    pub fn new_random() -> Self;
```

Comment:
Rust naming convention for constructor is:

Item | Convention
-- | --
General constructors | `new` or `with_more_details`
Conversion constructors | `from_some_other_type`

So I would rename `new_random()` to `with_random()`